gson/src/test/java/com/google/gson/JsonPrimitiveTest.java

```diff
@@ -300,10 +300,8 @@ public void testEqualsAcrossTypes() {
   @Test
   public void testEqualsIntegerAndBigInteger() {
     JsonPrimitive a = new JsonPrimitive(5L);
-    JsonPrimitive b = new JsonPrimitive(new BigInteger("18446744073709551621")); // 2^64 + 5
-    // Ideally, the following assertion should have failed but the price is too much to pay
-    // assertFalse(a + " equals " + b, a.equals(b));
-    assertWithMessage("%s equals %s", a, b).that(a.equals(b)).isTrue();
+    JsonPrimitive b = new JsonPrimitive(new BigInteger("18446744073709551621"));
+    assertWithMessage("%s not equals %s", a, b).that(a.equals(b)).isFalse();
```

Comment:
This may become irrelevant if we start using `EqualsTester`, but for now could you also check that `b.equals(a)` is false?

Reply:
We could leave it until we implement `EqualsTester`